module/zfs/dsl_crypt.c

```diff
@@ -1452,7 +1452,11 @@ spa_keystore_change_key_sync_impl(uint64_t rddobj, uint64_t ddobj,
 	 * Stop recursing if this dsl dir didn't inherit from the root
 	 * or if this dd is a clone.
 	 */
-	VERIFY0(dsl_dir_get_encryption_root_ddobj(dd, &curr_rddobj));
+	if (dsl_dir_get_encryption_root_ddobj(dd, &curr_rddobj) == ENOENT) {
```

Comment:
What happens if this returns something other than 0 or ENOENT?

Reply:
I should probably add a `VERIFY0()` after the check. We shouldn't be able to hit any other error case in syncing context, in theory. In reality, we can probably hit an IO error here which isn't checked in the `check` function. I'm not sure if it would be a good idea to do that check though, because that function might have to recurse through an arbitrary number of datasets. For now. I will just add a `VERIFY0()`